grpc/federation/cel/strings.go

```diff
@@ -0,0 +1,405 @@
+	"golang.org/x/text/cases"
+	"golang.org/x/text/language"
+)
+
+const StringsPackageName = "strings"
+
+var _ cel.SingletonLibrary = new(StringsLibrary)
+
+type StringsLibrary struct {
+	typeAdapter types.Adapter
+}
+
+func NewStringsLibrary(typeAdapter types.Adapter) *StringsLibrary {
+	return &StringsLibrary{
+		typeAdapter: typeAdapter,
+	}
+}
+
+func (lib *StringsLibrary) LibraryName() string {
+	return packageName(StringsPackageName)
+}
+
+func createStringsName(name string) string {
+	return createName(StringsPackageName, name)
+}
+
+func createStringsID(name string) string {
+	return createID(StringsPackageName, name)
+}
+
+func (lib *StringsLibrary) CompileOptions() []cel.EnvOption {
+	opts := []cel.EnvOption{}
+
+	for _, funcOpts := range [][]cel.EnvOption{
+		BindFunction(
+			createStringsName("clone"),
+			OverloadFunc(createStringsID("clone_string_string"), []*cel.Type{cel.StringType}, cel.StringType,
+				func(_ context.Context, args ...ref.Val) ref.Val {
+					return args[0].(types.String)
+				},
+			),
+		),
+		BindFunction(
+			createStringsName("compare"),
+			OverloadFunc(createStringsID("compare_string_string_int"), []*cel.Type{cel.StringType, cel.StringType}, cel.IntType,
+				func(_ context.Context, args ...ref.Val) ref.Val {
+					return args[0].(types.String).Compare(args[1].(types.String))
+				},
+			),
+		),
+		BindFunction(
+			createStringsName("contains"),
+			OverloadFunc(createStringsID("contains_string_string_bool"), []*cel.Type{cel.StringType, cel.StringType}, cel.BoolType,
+				func(_ context.Context, args ...ref.Val) ref.Val {
+					return types.Bool(strings.Contains(args[0].(types.String).Value().(string), args[1].(types.String).Value().(string)))
+				},
+			),
+		),
+		BindFunction(
+			createStringsName("containsAny"),
+			OverloadFunc(createStringsID("containsAny_string_string_bool"), []*cel.Type{cel.StringType, cel.StringType}, cel.BoolType,
+				func(_ context.Context, args ...ref.Val) ref.Val {
+					return types.Bool(strings.ContainsAny(args[0].(types.String).Value().(string), args[1].(types.String).Value().(string)))
+				},
+			),
+		),
+		BindFunction(
+			createStringsName("containsRune"),
+			OverloadFunc(createStringsID("containsRune_string_int_bool"), []*cel.Type{cel.StringType, cel.IntType}, cel.BoolType,
+				func(_ context.Context, args ...ref.Val) ref.Val {
+					str := args[0].(types.String).Value().(string)
+					r := rune(args[1].(types.Int).Value().(int64))
+					return types.Bool(strings.ContainsRune(str, r))
+				},
+			),
+		),
+		BindFunction(
+			createStringsName("count"),
+			OverloadFunc(createStringsID("count_string_string_int"), []*cel.Type{cel.StringType, cel.StringType}, cel.IntType,
+				func(_ context.Context, args ...ref.Val) ref.Val {
+					return types.Int(strings.Count(args[0].(types.String).Value().(string), args[1].(types.String).Value().(string)))
+				},
+			),
+		),
+		// func Cut(s, sep string) (before, after string, found bool)
```

Comment:
What is this comment? 🤔

Reply:
Sorry for the lack of explanation in the comments. I left a comment for the functions I determined to be impossible to implement. Except for `Cut`, the other comments are for functions that have arguments of types (like Go's `func` type) that cannot be received by CEL. As for `Cut`, I concluded it was impossible to implement because it needs to return two variables. 

Is there any way to return two variables in this case?

Reply:
We talked on this point directly and decided to update the comments to clarify why we cannot implement those functions.

Reply:
> Do we miss the Cut function?

Regarding "Cut," I found that it's feasible to implement, so I added the implementation.